### PR TITLE
feat: create a Loading Spinner component

### LINKED ISSUE #166: Create a `Loading Spinner` component
## Description

- The task is to create a `Loading Spinner` component.

## Criteria

- [ ] Display a background overlay and icon

### PR DESCRIPTION
**Pull request by HrRn**
## Overview

- I added a Loading Spinner component.

## Changes

- Created `loadingSpinner.tsx` component under `src/components/ui/`.
- Created `useLockBodyScroll.tsx` component under `src/hooks`.

## Screenshots or videos

https://github.com/Tascurator/tascurator-frontend/assets/77369174/ac63f680-ef77-403c-ab06-3cfbae3ba4bc

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [ ] Code readability and simplicity
- [ ] Follows best practices and coding standards
- [ ] Understandable and maintainable code

### CODE REVIEW

File: src/components/ui/loadingSpinner.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *     </main>
+ *     <Toaster />
+ *   </body>
+ * </html>
+ *
+ */
+
+export const LoadingSpinner = () => {
+  return (
+    <div className="fixed z-50 bg-black/50 w-full h-full flex justify-center items-center">
```
Comment (Akikaze1119) [inline src/components/ui/loadingSpinner.tsx]: Could you change 'fixed' to 'absolute'?
Example:
`fixed` => `absolute left-0 top-0`

File: src/components/ui/loadingSpinner.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *     <main className={'max-w-screen-sm min-h-screen w-full h-full bg-white px-6'}>
+ *       {children}
+ *     </main>
+ *     <Toaster />
+ *   </body>
+ * </html>
+ *
+ */
+
+export const LoadingSpinner = () => {
```
Comment (Akikaze1119) [inline src/components/ui/loadingSpinner.tsx]: Could you add isLoading prop for displaying the loading spinner separately?

▼Example
```export const LoadingSpinner = (isLoading: boolean) => {
  return (
    <>
      {isLoading && (
        <div className="absolute left-0 top-0 z-50 bg-black/50 w-full h-full flex justify-center items-center">
          <div className=" border-gray-200 h-12 w-12 animate-spin rounded-full border-4 border-t-primary" />
        </div>
      )}
    </>
  );
};
```

**Review by Akikaze1119 — CHANGES_REQUESTED** [review]
Thank you!
I'd like you to check my comments.

File: src/components/ui/loadingSpinner.tsx (hunk 2 above)
Comment (HrRn) [reply src/components/ui/loadingSpinner.tsx]: Thank you for your comments!
Following Akio's discussion, we made these changes:
- Added `useLockBodyScroll.tsx` in the hooks folder and implemented the logic there.
- Updated `loadingSpinner.tsx` to call useLockBodyScroll and use the isLoading prop to control the spinner's visibility.

**Review by Akikaze1119 — APPROVED** [review]
Thank you for improving codes!
LGTM ;)

**Pull request merged**